include/pybind11/common.h

```diff
@@ -225,6 +225,31 @@ struct buffer_info {
     ~buffer_info() {
         if (view) { PyBuffer_Release(view); delete view; }
     }
+
+    Py_buffer& as_pybuffer() const {
```

Comment:
I'd prefer to have this in the ``memoryview`` constructor, since it's breaking the pybind11 abstractions by exposing Python C API types. (and it's only ever used in ``memoryview``)